store/postgres/src/entities.rs

```diff
@@ -677,20 +697,72 @@ impl Table {
     }
 
     fn count_entities(&self, conn: &PgConnection) -> Result<u64, Error> {
-        match self {
+        let (count, subgraph) = match self {
             Table::Public(subgraph) => {
                 let count: i64 = public::entities::table
                     .filter(public::entities::subgraph.eq(subgraph.to_string()))
                     .count()
                     .get_result(conn)?;
-                Ok(count as u64)
+                (count as u64, subgraph)
             }
             Table::Split(entities) => {
                 let table = entities.table.clone();
                 let count: i64 = table.count().get_result(conn)?;
-                Ok(count as u64)
+                (count as u64, &entities.subgraph)
             }
-        }
+        };
+        let query = format!(
+            "
+            update subgraphs.entities
+            set data = data || '{{\"entityCount\": {{ \"data\": {}, 
+                                                      \"type\": \"Int\"}}}}'::jsonb
+            where entity='SubgraphDeployment'
+              and id = $1
+            ",
+            count
+        );
+        diesel::sql_query(query)
+            .bind::<Text, _>(subgraph.to_string())
+            .execute(conn)?;
+        Ok(count)
+    }
+
+    pub(crate) fn update_entity_count(
+        &self,
+        conn: &PgConnection,
+        subgraph: SubgraphDeploymentId,
+        count: i32,
+    ) -> Result<(), StoreError> {
+        let count_query = match self {
+            Table::Public(_) => format!(
+                "select count(*) from public.entities where subgraph = '{}'",
+                &subgraph
+            ),
+            Table::Split(entities) => format!("select count(*) from {}.entities", entities.schema),
+        };
+        // This query is pretty complicated since we try to cover both
+        // the case when we do not have an entityCount yet, and when we
+        // do have an entity count without roundtripping lots of data
+        // to the server. Note that `coalesce` only evaluates the arguments
+        // it needs, so that in the case where we already have an entityCount
+        // we do not needlessly cause a count of entities
+        let query = format!(
+            "
+            update subgraphs.entities
+            set data = data || (format('{{\"entityCount\":
+                                  {{ \"data\": %s, 
+                                    \"type\": \"Int\"}}}}',
```

Comment:
`entityCount` is a `BigInt` in the subgraph of subgraphs schema; and if we want the GraphQL resolver to return it as is (if it exists), it should probably be a `BigInt` here too.

Reply:
Good point. I changed that to `BigInt`